submit/services/filemanager.py

```diff
@@ -0,0 +1,420 @@
+
+    def upload_package(self, pointer: FileStorage) -> Tuple[dict, dict]:
+        """
+        Stream an upload to the file management service.
+
+        If the file is an archive (zip, tar-ball, etc), it will be unpacked.
+        A variety of processing and sanitization routines are performed, and
+        any errors or warnings (including deleted files) will be included in
+        the response body.
+
+        Parameters
+        ----------
+        pointer : :class:`FileStorage`
+            File upload stream from the client.
+
+        Returns
+        -------
+        dict
+            A description of the upload package.
+        dict
+            Response headers.
+        """
+        files = {'file': (pointer.filename, pointer, pointer.mimetype)}
+        return self.request('post', '/', files=files,
+                            expected_code=status.HTTP_201_CREATED)
+
+    def get_upload_status(self, upload_id: str) -> Tuple[dict, dict]:
+        """
+        Retrieve metadata about an accepted and processed upload package.
+
+        Parameters
+        ----------
+        upload_id : str
+            Unique long-lived identifier for the upload.
+
+        Returns
+        -------
+        dict
+            A description of the upload package.
+        dict
+            Response headers.
+
+        """
+        return self.request('get', f'/{upload_id}')
+
+    def add_files_to_package(self, upload_id: str, pointer: FileStorage) \
+            -> Tuple[dict, dict]:
+        """
+        Upload a file or package to an existing upload workspace.
+
+        If the file is an archive (zip, tar-ball, etc), it will be unpacked. A
+        variety of processing and sanitization routines are performed. Existing
+        files will be overwritten by files of the  same name. and any errors or
+        warnings (including deleted files) will be included in the response
+        body.
+
+        Parameters
+        ----------
+        upload_id : str
+            Unique long-lived identifier for the upload.
+        pointer : :class:`FileStorage`
+            File upload stream from the client.
+
+        Returns
+        -------
+        dict
+            A description of the upload package.
+        dict
+            Response headers.
+
+        """
+        return self.request('post', f'/{upload_id}', files={'file': pointer})
+
+    def delete_workspace(self, upload_id: str) -> Tuple[dict, dict]:
+        """
+        Delete the entire workspace.
+
+        Parameters
+        ----------
+        upload_id : str
+            Unique long-lived identifier for the upload.
+
+        Returns
+        -------
+        dict
+            An empty dict.
+        dict
+            Response headers.
+
+        """
+        return self.request('delete', f'/{upload_id}', expected_code=204)
+
+    def delete_all(self, upload_id: str) -> Download:
+        """
+        Delete all files in the workspace.
+
+        Does not delete the workspace itself.
```

Comment:
Potentially add a line about message back to the UI and/or confirmation? And in singular delete_file too.

Reply:
I guess the reason I didn't discuss UI considerations is that this is really just the service integration -- so that logic should all be in the controllers (next step).

Reply:
That makes sense. I wasn't sure whether there was an intermediary coming later where that would be more appropriate. :)